tools/lint/src/main/java/com/grab/lint/Sanitizer.kt

```diff
@@ -0,0 +1,65 @@
+package com.grab.lint
+
+import org.apache.commons.io.output.StringBuilderWriter
+import java.io.File
+
+class Sanitizer(
+    private val env: Env = Env.BazelEnv
+) {
+
+    fun sanitize(file: File): StringBuilderWriter {
+        val writer = StringBuilderWriter()
+
+        file.useLines { lines ->
+            lines.forEach { line ->
+                writer.appendLine(sanitize(line, rootRegex))
+            }
+        }
+        return writer
+    }
+
+    /**
+     * It seems we can't simply rely on PWD to find the execroot as Lint baseline output may contain sandbox realized paths in them which
+     * expands symlinks. For example, if the file is coming from another target's output it might contain a different sandbox root.
+     * Especially the sandbox number might be different.
+     *
+     * To solve this we check for sandbox path and prepare a [Regex] that can replace any sort of path sandbox or not.
+     */
+    private val rootRegex: Regex by lazy {
+        val pwd = env.pwd
+        val currDirName = File(pwd).name
+        val regex = "-sandbox/(.*?)/execroot/$currDirName".toRegex()
+        val sandboxDir = regex.find(pwd)?.groupValues?.firstNotNullOfOrNull { it.toIntOrNull() }
+        val pattern = when {
+            sandboxDir != null -> pwd.replace("/$sandboxDir/", "/(.*?)/")
+            else -> pwd
+        }
+        (pattern.removePrefix("/") + "/").toRegex()
+    }
+
+    private fun sanitize(line: String, calcExecRoot: Regex): String {
+        val target = if ("file=\"" in line) {
+            "file="
+        } else if ("message=\"" in line) {
+            "message="
+        } else {
+            return line
+        }
+
+        val suffix = if (line.endsWith("/>")) "/>" else if (line.endsWith(">")) ">" else ""
+        return FILE_PATH_REGEX.find(line)
+            ?.value
+            ?.replace("\"", "") // Remove "
+            ?.replace(System.getenv("TMPDIR") ?: "", "")
+            ?.dropWhile { char -> char == '.' || char == '/' } // Clean ../
+            ?.replace(calcExecRoot, "")
+            ?.let { fixedPath ->
+                "${line.split(target).first()}$target\"$fixedPath\"$suffix" // Retain indent and write file="updated path"
+            } ?: line
+    }
+
+
+    companion object {
+        private val FILE_PATH_REGEX = """"([^"]+)"""".toRegex()
```

Comment:
`FILE_PATH_REGEX` was added with `file=` in mind, just PATH_REGEX might be a better since it is used for both file and messages.